pkg/virt-controller/watch/vmi.go

```diff
@@ -225,6 +230,39 @@ func (c *VMIController) execute(key string) error {
 
 	logger := log.Log.Object(vmi)
 
+	for _, volume := range vmi.Spec.Volumes {
```

Comment:
I'd like to see this logic moved to the sync() function. It would also be nice if this logic was encapsulated into a function, so it reduces the amount of code within some of these larger functions

Reply:
Done